### DISCUSSION TITLE
How can a negative power value be modelised in a turbine

### DISCUSSION THREAD

**Discussion by Ooussamaa** [NatLabRockies/floris#643] [Q&A]
Hello Guys,

I have a Power SCADA DATA at my disposal, this DATA conaintes the power production for each turbine, Half of these valeus are positif and their order of magnitude is 1000, while the other half contains negatif values that have an order of magniture (10); these turbines are absorbing energy from the grid this is why they have negatif powers.

My question: How can we model the absorbation of energy from the grid by turbines on Floris, when I tryed, I got only a 1000 magnitude positive values.

For example:

SCADA                                             VS                                               Simulations on Floris using wake models: 
T1: 9000 KW                                    VS                                               T1: 9000 KW
T2: -23 KW                                       VS                                                T2: 8115 KW

Happy coding !

Thank you

**Comment by Bartdoekemeijer** [comment]
Hi @Ooussamaa,

In your example, it seems that T2 _should_ be producing 8 MW of energy, but in reality it is not producing any. Perhaps T2 is down for maintenance or due to another unforeseen reason. The small electricity consumption is probably because the turbine may still be yawing to mitigate loads and for other connectivity purposes.

Generally, when we are comparing FLORIS with SCADA, we remove such outlier (non-standard operation) data using [FLASC](https://github.com/NREL/flasc). In this manner, we circumvent the issue with trying to model this outlier behavior in FLORIS.

To answer your question, you could in theory model this behavior in FLORIS. Essentially, you have a turbine with a zero or very small negative power coefficient. You can create a new turbine definition that captures this behavior. However, you'll also have to model how the thrust coefficient behaves in such this non-producing operating condition. Generally, I'd suggest you consider what it is that you want to achieve in this analysis before diving deep in trying to model nontraditional turbine operation.

Hope this helps!

Cheers,
Bart

**Comment by Ooussamaa** [comment]
Hi Bart,

Thank you for answer, it will help me a lot !

In fact , I only gave the example of two turbines, acctually I am working on Seagreen (114 turbines) I can say Power DATA SCADA contains 70 negative small values for each combination (Ws, Wd), so I think it won't be due to maintenance operations, I found on the literature that it's due to the absorption of energy from grid by turbine.

Your proposition is really convinient, I will change the configuration of my turbine to model this behavior and much with SCADA.

Thank you Bart !

best reagrds !

Oussama
